src/sentry/integrations/bitbucket/repository.py

```diff
@@ -39,6 +39,33 @@ def get_repository_data(self, organization, config):
             config['name'] = repo['full_name']
         return config
 
+    def update_repository_data(self, repo, data):
+        """
+        Update repo in database if necessary. Return True if update happened,
+        False otherwise.
+        """
+        updated = False
+
+        new_name = data['identifier']
+        if repo.name != new_name:
+            repo.name = new_name
+            updated = True
+
+        new_url = u'https://bitbucket.org/{}'.format(data['name'])
```

Comment:
I think you'll need `full_name` here. For example one of my bitbucket repos requires my account in the path as well https://bitbucket.org/markstory/sphinx-contrib based on the bitbucket docs, `name` is just the repo name with no account prefix.

Reply:
`data['name']` actually is the `full_name` that comes back from the API - see the second-to-last line of `get_repository_data`.